bentoml/saved_bundle/templates.py

```diff
@@ -79,27 +79,28 @@
 COPY environment.yml requirements.txt setup.sh* bentoml-init.sh python_version* /bento/
 WORKDIR /bento
 
-# Execute permission for bentoml-init.sh
-RUN chmod +x /bento/bentoml-init.sh
+# copy over entrypoint scripts
+COPY docker-entrypoint.sh /usr/local/bin/
+
+# Execute permission for scripts
+RUN chmod +x /bento/bentoml-init.sh /usr/local/bin/docker-entrypoint.sh
 
 # Install conda, pip dependencies and run user defined setup script
 RUN if [ -f /bento/bentoml-init.sh ]; then bash -c /bento/bentoml-init.sh; fi
 
-# copy over model files
-COPY . /bento
+# copy over bundled_pip_dependencies
+COPY environment.yml bundled_pip_dependencies* /bento/
```

Comment:
there is no need to copy `environment.yml` right?

Reply:
`bundled_pip_dependencies` is only used for BentoML development - for installing local build of BentoML itself. This change will not benefit the end-user, nor the BentoML developers because the build cache will be invalidated if you are editing the BentoML source code.

Reply:
> there is no need to copy environment.yml right?
We need it because `bundled_pip_dependencies` may not exist. This prevents the COPY command from failing.

Reply:
> bundled_pip_dependencies is only used for BentoML development - for installing local build of BentoML itself. This change will not benefit the end-user, nor the BentoML developers because the build cache will be invalidated if you are editing the BentoML source code.

I'm aware of that. You will find it actually benefits when `bundled_pip_dependencies` exist. Also, the main purpose of this PR is to make sure that the artifact coping is in the top layer.